docs/source-pytorch/deploy/production_basic.rst

```diff
@@ -71,10 +71,31 @@ When you need to add complicated pre-processing or post-processing logic to your
 ****************************
 Enable distributed inference
 ****************************
-By using the predict step in Lightning you get free distributed inference
-
+By using the predict step in Lightning you get free distributed inference using :class:`~pytorch_lightning.callbacks.prediction_writer.BasePredictionWriter`.
 
 .. code-block:: python
 
-    trainer = Trainer(devices=8, accelerator="gpu")
-    predictions = trainer.predict(model, data_loader)
+    import torch
+    from pytorch_lightning.callbacks import BasePredictionWriter
+
+
+    class CustomWriter(BasePredictionWriter):
+        def __init__(self, output_dir, write_interval):
+            super().__init__(write_interval)
+            self.output_dir = output_dir
+
+        def write_on_epoch_end(self, trainer, pl_module, predictions, batch_indices):
+            # this will create N files in `output_dir` each containing the predictions of it's respective rank
```

Comment:
This mentions N, but not what N is. I suggest replacing it for number of processes.

Reply:
done!